webpack/scenes/Subscriptions/Manifest/CdnConfigurationForm.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+
+import {
+  ActionGroup,
+  Alert,
+  Button,
+  Form,
+  FormAlert,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  TextInput,
+} from '@patternfly/react-core';
+
+import { translate as __ } from 'foremanReact/common/I18n';
+
+import EditableTextInput from '../../../components/EditableTextInput';
+
+import {
+  selectUpdatingCdnConfiguration,
+} from '../../Organizations/OrganizationSelectors';
+
+import { updateCdnConfiguration } from '../../Organizations/OrganizationActions';
+import './CdnConfigurationForm.scss';
+
+const CdnConfigurationForm = (props) => {
+  const {
+    contentCredentials,
+    cdnConfiguration,
+  } = props;
+
+  const dispatch = useDispatch();
+
+  const [url, setUrl] = useState(cdnConfiguration.url);
+  const [username, setUsername] = useState(cdnConfiguration.username);
+  const [password, setPassword] = useState(null);
+  const [organizationLabel, setOrganizationLabel] =
+    useState(cdnConfiguration.upstream_organization_label);
+  const [sslCaCredentialId, setSslCaCredentialId] = useState(cdnConfiguration.ssl_ca_credential_id);
+  const updatingCdnConfiguration = useSelector(state => selectUpdatingCdnConfiguration(state));
+
+  const editPassword = (value) => {
+    if (value === null) {
+      setPassword('');
+    } else {
+      setPassword(value);
+    }
+  };
+
+  const hasPassword = (cdnConfiguration.password_exists && password === null)
```

Comment:
For this condition,
```js
cdnConfiguration.password_exists && password === null
```
my form is still showing invalid even though it's valid.  (I had an existing cdn configuration saved already.)
When I enter a password in the box and click :ballot_box_with_check:, the alert disappears.  I think the validation needs to somehow check `hasPassword` and not just `password`..

Reply:
nice catch - will fix